Train word2vec embeddings on dataset

bumbledbees opened the pull request:
- add src/word2vec.py, which uses `gensim` to tune the Google News word2vec embeddings to our dataset.
- add data/w2v-headline-embeddings.bin, which contains the learned embedding values.
- add data/get-embeddings.sh to download the huge pre-trained embeddings file without committing it to the source tree.
- fixed an inaccurate type hint in src/data_normalize.py.
- add `gensim` to requirements.txt.

bumbledbees committed:
Train word2vec embeddings on dataset

- add src/word2vec.py, which uses `gensim` to tune the Google News
  word2vec embeddings to our dataset.
- add data/w2v-headline-embeddings.bin, which contains the learned
  embedding values.
- add data/get-embeddings.sh to download the huge pre-trained embeddings
  file without committing it to the source tree.
- fixed an inaccurate type hint in src/data_normalize.py.
- add `gensim` to requirements.txt.

tsunami935 commented on src/word2vec.py:
```diff
@@ -0,0 +1,57 @@
+    print('Learning embeddings...')
+    model.train(headlines,
+                total_examples=model.corpus_count,
+                epochs=model.epochs)
```
All this does is add embeddings for words w/o embeddings yet and takes existing embeddings from GOOGLE_WORD2VEC w/o updating them. I feel that we might want to update the pre-trained embeddings as well (i.e. for words like 'trump').

tsunami935 reviewed (commented):
Looking at the source code, it does not look like the pre-trained weights are being fine-tuned. It just looks we're filtering from the pre-trained embeddings for words in our dataset, then adding embeddings for words in our dataset that don't have pre-trained embeddings yet. https://radimrehurek.com/gensim/models/keyedvectors.html#gensim.models.keyedvectors.KeyedVectors.intersect_word2vec_format

bumbledbees committed:
Add code to specify w2v training params

- moved core training functionality out of main() in word2vec.py and
  into a new function, w2v_train().
- ensured value of lockf in .intersect_word2vec_format() is being
  provided.
- added ability to provide command-line arguments to word2vec.py
- added a bash script to train embeddings with various lockf values.
- added some extra entries to .gitignore
